### DISCUSSION TITLE
Homebrew update/install fails with SHA mismatch

### DISCUSSION THREAD

**Discussion by alexhallam** [Homebrew/discussions#3266] [Writing Formulae/Casks] [answered]
I recently updated my package. I did the steps outlined in the contribution guide.

```
brew uninstall --force foo
brew install --build-from-source foo
brew test foo
brew audit --strict foo
brew style foo
```

The problem is that I had a user report that there is a sha mismatch. 

https://github.com/alexhallam/tv/issues/129

I am guessing that this is due to the bottles. So I have some questions.

1. Is the problem a sha associated with a bottle
2. If the problem is with a bottle sha am I responsible for getting the new shas from the HomeBrew Bot in there? 

From the docs it looks like I need to add something like `revision 1` somewhere in the formula if I want to refresh the bottles, but I am not sure where to do that.

Here is my formula. https://github.com/Homebrew/homebrew-core/blob/master/Formula/tidy-viewer.rb

**Comment by SMillerDev** [comment]
If they can build it from source their cached version is wrong. There's nothing brew can do about that

**Comment by gromgit** [comment, marked as answer]
> I am guessing that this is due to the bottles.

No, the problem is this (from your user's issue report):
```
$ brew update && brew upgrade tidy-viewer
Already up-to-date.
==> Upgrading 1 outdated package:
alexhallam/tidy-viewer/tidy-viewer 1.4.3 -> 1.4.5
```

They're upgrading from your personal tap, not Homebrew core, and your personal tap's formula indeed has a SHA256 source tarball mismatch.
